lib/spack/spack/test/llnl/util/tty/log.py

```diff
@@ -396,6 +396,11 @@ def mock_shell_v_v_no_termios(proc, ctl, **kwargs):
 def test_foreground_background_output(
         test_fn, capfd, termios_on_or_off, tmpdir):
     """Tests hitting 'v' toggles output, and that force_echo works."""
+    if (sys.version_info >= (3, 8) and sys.platform == 'darwin'
+        and termios_on_or_off == no_termios):
+
+        return
```

Comment:
It might be better to do this as a `pytest.mark.skipif` instead. Can you have multiple?

Reply:
You can have multiple, but I don't know how to do it based on the parameterization (i.e. I can check for the python version and platform, but I'm not sure how I would access `termios_on_or_off`).